whatshap/variants.py

```diff
@@ -574,11 +582,12 @@ def realign(
         splitted_strings,
     ):
         """
-        Realign a read to the two alleles of a single variant.
+        Realign a read to the two alleles (or to the valid_alleles if it is not None) of a single variant.
         i and consumed describe where to split the cigar into a part before the
         variant position and into a part starting at the variant position, see split_cigar().
 
         variant -- VcfVariant
+        valid_alleles -- list of valid alleles
```

Comment:
The variable name `valid_alleles` is used for different things in different places, which makes the code a bit harder to read than it needs to be. I think it is a list of `Genotype` objects above and just a single `Genotype` here. Would make it easier if it we used different names, e.g. `valid_alleles` here and `valid_allele_list` above.

Reply:
I start to use `predicted_genotypes` and `predicted_alleles`.